pyramus/src/main/java/fi/otavanopisto/pyramus/json/students/EditStudentJSONRequestController.java

```diff
@@ -523,7 +531,34 @@ else if (phoneId > 0 && number != null) {
           ApplicationUtils.deleteApplication(application);
         }
       }
+      
+      // Student card
+      
+      StudentCard studentCard = studentCardDAO.findByStudent(student.getId());
+
+      Date expiryDate = null;
+      
+      // Set expiry date automatically same as study end date or study time end
+      if (student.getStudyEndDate() != null) {
```

Comment:
Tämä voisi olla ihan validi lähestymistapa, en nyt äkkisältään osaa ajatella miten kaikki tulkinnat tulisi toimimaan, jos päättymispäivä tallennettaisiin näin.

Itse ajattelin alunperin, että päättely tehtäisiin vasta listaamisvaiheessa, jolloin tässä tallennettaisiin aina vain tämä "kovakoodattu" aikaraja. Sen huono puoli on siinä, että tulkinta tulee olemaan vähän hankalaa, mutta hyvä puoli siinä, että nämä eri päivämäärät olisi aina erikseen tallessa. Mutta joo, en nyt osaa ajatella kaikkia vaikutuksia tähän päättelylogiikkaan niin voinee olla näinkin.. :D